Feature/559 schedule view filter by course prefix

Urjikoo opened the pull request:
## Types of changes
<!--
  What types of changes does your code introduce? Put an `x` in all the boxes
  that apply:
-->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality
      to change)

## Checklist:
<!--
  Go over all the following points, and put an `x` in all the boxes that apply.
-->
- [x] I have run `eslint` on the code
- [ ] I have added JSDoc for all of my code (where applicable)
- [x] I have added tests to cover my changes.

## Priority:
- [x] Normal <!-- New piece of functionality -->
- [ ] High <!-- Critical bug requiring urgent review -->

## Related Issues:
<!--
  Add the issue number in below that this PR is intended to address.
 #559 
  Also mention any issues that this PR is related to, and if possible,
  provide more information regarding the relationship to the issues in
  question (i.e does the PR fix the issue, is it designed to fix another bug
  that is similar to the issue etc.)
-->
Fixes #559 

<!--
  Attribution:
  PR Template adapted from: https://github.com/h5bp/html5-boilerplate/blob/master/.github/PULL_REQUEST_TEMPLATE.md
-->

Urjikoo committed:
added course prefix filter buttons that fades the session block

Urjikoo committed:
Course filter buttons added to fade courseprefix blocks.
when the course filter button is selected it fades the sessionblock for the course and the courselisting button inside the block will be disabled.

Urjikoo committed:
The filter test to check whether the courselisting button is disabled is added.
the test checks if the courselisting button is disabled by checking if the popover is disabled/null after the filter button is clicked.

Urjikoo committed:
test updated to check if the coursefilter button successfully disables the courselisting button when selected.
when the courseprefix button is active, the courselisting button is disabled that there is no popover when courselisting button is selected.

jonseitz reviewed (changes requested):
Overall you've done a really nice job with this! The functionality itself is working as expected. There are a few things that need to be cleaned up in the code, mostly spacing things that `eslint` didn't catch and documentation, along with a few optimization things that have to do with the peculiarities of React's render cycles. 

I've tried to provide explanations and example code where relevant, and I'm happy to chat about these notes in more depth during standup or on slack.

Urjikoo committed:
The spacing and code clean up completed.
the styling for jsdoc along with some code definitions updated to include spacing and separate lines.

Urjikoo commented on src/client/components/pages/Schedule/SchedulePage.tsx:
```diff
@@ -61,7 +77,37 @@ const SchedulePage: FunctionComponent = () => {
+    const prefixObj = prefixes.find((p) => p.prefix === prefix);
+    return prefixObj.active;
+  }, [prefixes]);
+  const prefixToggleButtons = prefixes.map((prefixObj) => (
+    <PrefixButton
+      alt="Course Filter Button"
+      prefix={prefixObj.prefix}
+      variant={VARIANT.BASE}
+      key={prefixObj.prefix}
+      onClick={() => togglePrefix(prefixObj.prefix)}
+    >
+      {prefixObj.prefix}
+    </PrefixButton>
+  ));
```
I did render the `prefixToggleButtons` inside the return block by passing it in as {prefixToggleButtons} because it wouldn't let me declare it in return block. so should i declare it right before return block ?

jonseitz replied:
Right, I'm suggesting that instead of declaring a variable and passing it in, you can just call the `.map` function directly in the component, i.e.:

```tsx
return (
  <>
    <VerticalSpace>
      {prefixes.map((prefixObj) => (
        <PrefixButton
          alt="Course Filter Button"
          prefix={prefixObj.prefix}
          variant={VARIANT.BASE}
          key={prefixObj.prefix}
          onClick={() => togglePrefix(prefixObj.prefix)}
        >
          {prefixObj.prefix}
        </PrefixButton>
      ))}
...
```

Urjikoo committed:
updated prefixbutton rendered in the verticalspace component.
instead of declaring prefixbutton outside of return block  and passing it in, it was mapped and rendered in the component

jonseitz reviewed (approved):
These updates all look great! I also re-ran the GitHub Action tests and those are coming up green now (not sure why it was failing in the first place, but that's a mystery for another day...)

This is all set to merge!
